src/main/kotlin/com/figure/gradle/semver/internal/tasks/GenerateVersionFileTask.kt

```diff
@@ -7,23 +7,19 @@
 
 package com.figure.gradle.semver.internal.tasks
 
-import com.figure.gradle.semver.internal.semverDebug
 import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.InputDirectory
-import org.gradle.api.tasks.PathSensitive
-import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 import org.gradle.work.DisableCachingByDefault
-import java.io.File
 
-@DisableCachingByDefault(because = "Not worth caching")
+//@DisableCachingByDefault(because = "Not worth caching")
 internal abstract class GenerateVersionFileTask : DefaultTask() {
 
-    @get:InputDirectory
-    @get:PathSensitive(PathSensitivity.ABSOLUTE)
-    internal abstract val buildDir: Property<File>
+    @get:OutputFile
```

Comment:
I think this is the meat here - instead of saying 'give a directory as input' this is saying 'set a file path that should be the output of this job'. The task registration sets up the file location, and then the task itself creates it.